src/cockpit/samples.py

```diff
@@ -39,6 +43,11 @@ def read_int_file(rootfd: int, statfile: str, default: Optional[int] = None, key
 
     try:
         data = os.read(fd, 1024)
+    except OSError as e:
+        # cgroups can disappear between the open and read
+        if e.errno != errno.ENODEV:
+            logger.warning('Failed to read %s: %s', statfile, e)
```

Comment:
Pushing back in the other direction: if this is really a legit race condition (ie: we expect that groups might disappear sometimes, while scanning) then actually `.warning()` is too much.  `.debug()` makes more sense to me.

Reply:
ENODEV is the expected case, which is now quiet. This is more about EIO, EBADF, etc., which are unexpected situations that should cause a test failure IMHO (unexpected message), and also make this possible to spot and debug.

Reply:
Ah right!  Sorry that I read it wrong.